engine/framework/master.go

```diff
@@ -670,6 +682,42 @@ func (m *DefaultBaseMaster) IsMasterReady() bool {
 	return m.workerManager.IsInitialized()
 }
 
+// Exit implements BaseMaster.Exit
+func (m *DefaultBaseMaster) Exit(ctx context.Context, exitReason ExitReason, err error, extMsg string) error {
+	// Set the errCenter to prevent user from forgetting to return directly after calling 'Exit'
+	// keep the original error in errCenter if possible
+	defer func() {
+		if err == nil {
+			err = derror.ErrWorkerFinish.FastGenByArgs()
+		}
```

Comment:
Can we return a `ErrWorkerCancel` if exit reason is `ExitReasonCancelled`, with which the executor runtime can detect whether the worker is finished or canceled.